_posts/2023-02-22-#25038.md

```diff
@@ -7,18 +7,234 @@ authors: [glozow]
+	* [Eltoo Pinning](https://gist.github.com/instagibbs/60264606e181451e977e439a49f69fe1)
+
+A larger list of RBF limitations was discussed last year on [this mailing list
+post](https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-January/019817.html) and
+[gist](https://gist.github.com/glozow/25d9662c52453bd08b4b4b1d3783b9ff). Various proposals to amend or rewrite RBF
+policy were discussed on those threads. The discussions resulted in this v3 proposal to address most issues. Additional
+RBF improvements (which do not conflict with v3) are also being considered.
+
+### V3 Policy
+
+The philosophy behind [v3](https://bitcoinops.org/en/topics/version-3-transaction-relay/) is that RBF limitations stem
+from current mempool policy being too permissive in the possible package topologies it accepts. [Package
+limits](https://github.com/bitcoin/bitcoin/blob/master/doc/policy/mempool-limits.md) exist to limit the worst-case
+computational complexity of mempool operations such as block template assembly and new transaction validation. However,
+current limits still allow huge clusters of thousands of transactions.
+
+<!-- FIXME: add photo -->
+
+With these possible transaction graphs, it isn't feasible to add a comparison of incentive compatibility scores, as
+calculating them could involve iterating through the entire mempool. Permissive package limits are also what makes Rule
+3 pinning possible; when two counterparties share a transaction, they are both able to bring in descendants and other
+transactions to influence the incentive compatibility of their shared transaction.
+
+On the other hand, most protocols involving untrusted counterparties don't require complex transaction graphs. Usually,
+a 1-parent-1-child package is sufficient: the presigned transaction plus a child attaching fees at broadcast time.
+
+V3 rules are as follows:
+- A v3 transaction signals replaceability, even if it does not signal BIP125 replaceability.
+- Any descendant of an unconfirmed V3 transaction must also be V3.
+- A V3 transaction's unconfirmed ancestors must all be V3.
+- A V3 transaction cannot have more than 1 unconfirmed descendant.
+- A V3 transaction that has an unconfirmed V3 ancestor cannot be larger than 1000 virtual bytes.
+- A V3 transaction cannot have more than 1 unconfirmed ancestor.
+
+Notice a few nice properties of v3 packages and transactions (from most obvious to least obvious):
+- A v3 transaction in mempool cannot have more than 1 descendant.
+- A v3 transaction in mempool cannot have more than 1 ancestor.
+- When a v3 transaction is being replaced, it has at most 1 descendant. That descendant adds no more than 1000vB to what needs to be evicted.
+- When the descendant of a v3 transaction is being replaced, there must be exactly 1 replacement transaction and 1 to-be-replaced transaction.
+- When a v3 transaction is being replaced, it is the sponsor of either 0 or 1 0-fee parents.
```

Comment:
It just seems like a more specific case of "cannot have more than 1 ancestor", but I'm not sure why the 0-fee use case is so important here?

Reply:
I was mostly confused by the "0-fee", because you could essentially copy/paste that sentence replacing 0 (in "0-fee") with any number and it would still be true. If I understand your intent correctly, how about:

```suggestion
- When a v3 transaction is being replaced, it is the sponsor of either 0 or 1 (but never more than that) parents that would fall below the minimum feerate without the child sponsoring it.
```